### ISSUE TITLE
__IMPOSSIBLE__ at src/full/Agda/TypeChecking/Rules/LHS.hs:1552:19 in Agda-2.6.2

### ISSUE THREAD

**Issue by JasonGross** [agda/agda#6087]
* [x] My report includes the version of Agda (and Emacs, GHC, and OS versions where relevant).
* [ ] The issue can be reproduced with the _latest version_ of Agda (either [released](https://wiki.portal.chalmers.se/agda/Main/Download) or `master` branch). (I have only checked with 2.6.2, but it seems this issue has not been reported.  If it is not reproducible with the latest Agda, I apologize; please close this bug, and the next person who hits it will hopefully now find it on the issue tracker; if there's not a test-case for it in the test-suite, then this one can be added.)
* [x] I have supplied all information so the bug can be easily reproduced (in particular, the reproducer is a complete Agda file).

```agda
module foo where
open import Agda.Primitive using (Level; lzero; lsuc; _⊔_; Set) public
record _×_ {l : Level} (A B : Set l) : Set l where
  constructor _,_
  field fst : A
  field snd : B
infixl 0 _,_
infixl 6 _×_
data _≡_ {l} {A : Set l} (a : A) : A -> Set l where
  refl : a ≡ a
infix 1 _≡_
sub : ∀ {l} {A : Set l} {m} (B : A -> Set m) {a1 a2 : A} -> a1 ≡ a2 -> B a1 -> B a2
sub B refl b = b
cast : ∀ {l} {A B : Set l} -> (A ≡ B) -> (A -> B)
cast refl a = a
record _×≡_ {l} {A B : Set l} (ab1 ab2 : A × B) : Set l where
  constructor _,≡_
  private module ab1 = _×_ ab1
  private module ab2 = _×_ ab2
  field fst≡ : ab1.fst ≡ ab2.fst
  field snd≡ : ab1.snd ≡ ab2.snd
  wrap : ab1 ≡ ab2
  wrap with fst≡ | snd≡
  ...     | refl  | refl  = refl
infixl 0 _,≡_
module _,≡_ {l} {A} {B} {ab1} {ab2} ff ss = _×≡_ (_,≡_ {l} {A} {B} {ab1} {ab2} ff ss)
record Object (l : Level) : Set (lsuc l) where
  field Obj : Set l
  field Arr : Obj -> Obj -> Set l
  field comp : ∀ {a1 a2 a3} -> (Arr a1 a2 × Arr a2 a3) -> Arr a1 a3
record Object= {l} (A B : Object l) : Set (lsuc l) where
  private module A = Object A
  private module B = Object B
  field Obj= : A.Obj ≡ B.Obj
  field Arr= : ∀ {a1 a2} -> A.Arr a1 a2 ≡ B.Arr (cast Obj= a1) (cast Obj= a2)
  comp=-motive : ? -> ?
  comp=-motive = (λ (Arra1a2 , Arra2a3 , Arra1a3) -> (Arra1a2 × Arra2a3) -> Arra1a3)
  {- An internal error has occurred. Please report this as a bug.
Location of the error: __IMPOSSIBLE__, called at src/full/Agda/TypeChecking/Rules/LHS.hs:1552:19 in Agda-2.6.2-JhaFTuyX5eV47ZrlsAQi0M:Agda.TypeChecking.Rules.LHS
-}
  uncurry-Arr= = λ {a1} {a2} {a3} -> _,≡_.wrap (_,≡_.wrap (Arr= {a1} {a2}) (Arr= {a2} {a3})) (Arr= {a1} {a3})
  field comp= : ∀ {a1 a2 a3} -> sub comp=-motive (uncurry-Arr= {a1} {a2} {a3})
                                      A.comp ≡ B.comp
```
This is
https://github.com/agda/agda/blob/246a229faea2ef2fa53caf491ff8a2d72dd7510c/src/full/Agda/TypeChecking/Rules/LHS.hs#L1552
in `isDataOrRecordType`

**Comment by jespercockx** [comment]
Thanks for reporting this! I cannot reproduce this on the latest `master`, so it seems this has already been fixed. We're aiming for a release of 2.6.3 by the end of October, so hopefully you won't have to wait much longer.

**Comment by andreasabel** [comment]
The internal error can be reproduced in 2.6.1 and 2.6.2, but it was already fixed in 2.6.2.1.
